lsstseries/analysis/structurefunction2.py

```diff
@@ -0,0 +1,115 @@
+from scipy.stats import binned_statistic
+import numpy as np
+
+
+def calc_sf2(time, flux, err, bins, band, band_to_calc=None):
+    """Compute structure function squared on one or many bands
+
+    Parameters
+    ----------
+    time : `numpy.ndarray` (N,)
+        Array of times when measurements were taken.
+    flux : `numpy.ndarray` (N,)
+        Array of flux/magnitude measurements.
+    err : `numpy.ndarray` (N,)
+        Array of associated flux/magnitude errors.
+    bins : `numpy.ndarray` (N,)
+        Array of time bins in which to calculate structure function
+    band : `numpy.ndarray` (N,)
+        Array of associated band labels,
+    band_to_calc : `str` or `list` of `str`
+        Bands to calculate StetsonJ on. Single band descriptor, or list
+        of such descriptors.
+
+    Returns
+    -------
+
+    Notes
+    ----------
+    In case that no value for `band_to_calc` is passed, the function is
+    executed on all available bands in `band`.
+    """
+
+    unq_band = np.unique(band)
+
+    if band_to_calc is None:
+        band_to_calc = unq_band
+    if isinstance(band_to_calc, str):
+        band_to_calc = [band_to_calc]
+
+    assert hasattr(band_to_calc, "__iter__") is True
+
+    sf2 = {}
+    for b in band_to_calc:
+        if b in unq_band:
+            mask = band == b
+            times = time[mask]
+            fluxes = flux[mask]
+            errors = err[mask]
+            sf2[b] = _sf2_single(times, fluxes, errors, bins)
+        else:
+            sf2[b] = np.nan
+    return sf2
+
+
+def _sf2_single(times, fluxes, errors, bins):
+    """Calculate structure function squared
+
+    Calculate structure function squared from the avaliable data. This is
```

Comment:
avaliable -> available

Reply:
Fixed. I am still searching for a spell checker that I could use only on the comments....